frontend/www/js/omegaup/components/problem/BaseList.vue

```diff
@@ -0,0 +1,295 @@
+                  hrefForProblemTag(currentTags, tag.name).includes('Level')
+                    ? 'custom-badge-quality'
+                    : ''
+                } m-1 p-2`"
+                :href="hrefForProblemTag(currentTags, tag.name)"
+                >{{
+                  Object.prototype.hasOwnProperty.call(T, tag.name)
+                    ? T[tag.name]
+                    : tag.name
+                }}</a
+              >
+            </td>
+            <td
+              v-if="problem.quality !== null"
+              class="text-center tooltip_column"
+            >
+              <span
+                v-tooltip="
+                  `${ui.formatString(T.wordsOutOf4, {
+                    Score: problem.quality.toFixed(1),
+                  })}`
+                "
+              >
+                {{ QUALITY_TAGS[Math.round(problem.quality)] }}
+              </span>
+            </td>
+            <td v-else class="text-right">—</td>
+            <td v-if="problem.difficulty !== null" class="text-center">
+              <span
+                v-tooltip="
+                  `${ui.formatString(T.wordsOutOf4, {
+                    Score: problem.difficulty.toFixed(1),
+                  })}`
+                "
+              >
+                {{ DIFFICULTY_TAGS[Math.round(problem.difficulty)] }}
+              </span>
+            </td>
+            <td v-else class="text-center">—</td>
+            <td class="text-right">
+              {{ (100.0 * problem.ratio).toFixed(2) }}%
+            </td>
+            <td v-if="loggedIn" class="text-right">
+              {{ problem.score.toFixed(2) }}
+            </td>
+            <td class="text-right">{{ problem.points.toFixed(2) }}</td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+    <div class="card-footer">
+      <omegaup-common-paginator
+        :pager-items="pagerItems"
+      ></omegaup-common-paginator>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import { omegaup } from '../../omegaup';
+import T from '../../lang';
+import { types } from '../../api_types';
+import * as ui from '../../ui';
+
+import common_Paginator from '../common/Paginatorv2.vue';
+import common_SortControls from '../common/SortControls.vue';
+
+import 'v-tooltip/dist/v-tooltip.css';
+import { VTooltip } from 'v-tooltip';
+
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
+import {
+  faEyeSlash,
+  faMedal,
+  faExclamationTriangle,
+  faBan,
+} from '@fortawesome/free-solid-svg-icons';
+library.add(faEyeSlash, faMedal, faExclamationTriangle, faBan);
+
+@Component({
+  components: {
+    FontAwesomeIcon,
+    'omegaup-common-paginator': common_Paginator,
+    'omegaup-common-sort-controls': common_SortControls,
+  },
+  directives: {
+    tooltip: VTooltip,
+  },
+})
+export default class ProblemBaseList extends Vue {
```

Comment:
nit:

```suggestion
export default class BaseList extends Vue {
```

hay que mantener consistencia entre los nombres de los componentes y los nombres de los archivos